### PR TITLE
docs: Improve friendliness of the dev docs

### PR DESCRIPTION
**Pull request by draychev**
With influx of new gophers coming to the repo we need to update the dev docs to help onboarding to the repo easier.
This PR removes a few stale TODOs, also makes the list of directories collapsable. Adding our guiding principle on how we think about the code base.

Please describe the motivation for this PR and provide enough
information so that others can review it.

Please mark with X for applicable areas.

- [ ] New Functionality
- [x] Documentation
- [ ] Install
- [ ] Control Plane
- [ ] CLI Tool
- [ ] Certificate Management
- [ ] Networking
- [ ] Metrics
- [ ] SMI Policy
- [ ] Security
- [ ] Tests / CI System
- [ ] Other

Please answer the following questions with yes/no.

- Does this change contain code from or inspired by another project? If so, did you notify the maintainers and provide attribution? - no

### CODE REVIEW

File: docs/development_guide.md [hunk 1]
```diff
@@ -1,24 +1,16 @@
 # Open Service Mesh Development Guide
 
-Welcome to the Open Service Mesh development guide!
+Thank you for joining us on a journey to build an SMI-native lightweight service mesh. The first of our [core principles](https://github.com/openservicemesh/osm#core-principles) is to create a system, which is "simple to understand and contribute to." We hope that you would find the source code easy to understand. If not - we invite you to help us fulfil this principle. There is no PR too small!
```
Comment (ksubrmnn) [inline docs/development_guide.md:3]: *fulfill

Comment (ksubrmnn) [reply docs/development_guide.md:3]: I'd keep the Welcome message and add your part underneath it

File: docs/development_guide.md [hunk 2]
```diff
@@ -1,24 +1,16 @@
...
 
-This document will help you build and run Open Service Mesh from source.
-More information about running the demo included in this repo is
-in [/demo/README.md](../demo/README.md).
-The OSM software design is discussed
-in detail in [DESIGN.md](/DESIGN.md).
+To understand *what* Open Service Mesh does - take it for a spin and kick the tires. Install it on your Kubernetes cluster by following [this guide](./example/README.md).
 
-## Table of contents
+To get a deeper understanding of how OSM does what it does - take a look at the detailed [software design](../DESIGN.md).
```
Comment (shashankram) [inline docs/development_guide.md:8]:
```suggestion
To get a deeper understanding of how OSM does what it does - take a look at the detailed [software design](/DESIGN.md).
```

File: docs/development_guide.md [hunk 3]
```diff
@@ -61,46 +53,24 @@ as a birds-eye view of where the different components are located.
...
 
-TODO:
- - Describe all the environment variables.
- - how to enable various levels of debugging - mostly tracing.
- - Describe the format of the debug messages.
- - How to enable the debug server & how it could be useful.
+The OSM packages rely on many external Go libraries.
 
-### Go
+Take a peek at the [go.mod](../go.mod) file to see all dependencies.
```
Comment (shashankram) [inline docs/development_guide.md:72]:
```suggestion
Take a peek at the [go.mod](/go.mod) file to see all dependencies.
```

File: docs/development_guide.md [hunk 4]
```diff
@@ -114,6 +84,13 @@ More notable:
   - `make go-vet` - same as `go vet ./...`
 
 ## Create Environment Variables
+
+The OSM repo relies on environment variables to make it usable on your localhost. The root of the repository contains a file named `.env.example`. Copy the contents of this file into `.env`
+```bash
+cat .env.example > .env
+```
+Tha various envirnoment variables are documented in the `.env` file itself. Modify the variables in `.env` to suite your environment.
```
Comment (shashankram) [inline docs/development_guide.md:93]:
```suggestion
The various envirnoment variables are documented in the `.env.example` file itself. Modify the variables in `.env` to suite your environment.
```

File: docs/development_guide.md [hunk 5]
```diff
@@ -185,10 +146,7 @@ The different chart templates are used as follows:
 - `prometheus-*.yaml` chart templates are used to deploy a Prometheus instance when the metrics stack is enabled
 - `zipkin-*.yaml` chart templates are used to deploy a Zipkin instance when Zipkin tracing is enabled
 
+## Custom Resource Definitions
```
Comment (shashankram) [inline docs/development_guide.md:150]: Should be a subsection of `Helm charts`

File: docs/development_guide.md [hunk 6]
```diff
@@ -61,46 +53,24 @@ as a birds-eye view of where the different components are located.
     - `smi/` - SMI client, informer, caches and tools
     - `tests/` - test fixtures and other functions to make unit testing easier
     - `trafficpolicy/` - SMI related types
+</details>
 
+The Open Service Mesh controller is written in Go.
+It relies on the [SMI Spec](https://github.com/servicemeshinterface/smi-spec/).
+OSM leverages [Envoy proxy](https://github.com/envoyproxy/envoy) as a data plane and Envoy's [XDS v3](https://github.com/envoyproxy/go-control-plane) protocol.
```
Comment (eduser25) [inline docs/development_guide.md:61]:
```suggestion
OSM leverages [Envoy proxy](https://github.com/envoyproxy/envoy) as a data plane and Envoy's [XDS v3](https://www.envoyproxy.io/docs/envoy/latest/api-v3/api) protocol, which is offered in Go by [go-control-plane](https://github.com/envoyproxy/go-control-plane).
```
Just suggesting

File: docs/development_guide.md (hunk 5 above)
Comment (draychev) [reply docs/development_guide.md]: I added another `#` to indent under `Helm charts`
Thank you!

**Comment by shashankram** [comment]
> Please answer the following questions with yes/no.
> 
> * Does this change contain code from or inspired by another project? If so, did you notify the maintainers and provide attribution?

@draychev please answer this question in the PR template.

File: docs/development_guide.md [hunk 7]
```diff
@@ -1,24 +1,17 @@
...
 
-This document will help you build and run Open Service Mesh from source.
-More information about running the demo included in this repo is
-in [/demo/README.md](../demo/README.md).
-The OSM software design is discussed
-in detail in [DESIGN.md](/DESIGN.md).
+To understand *what* Open Service Mesh does - take it for a spin and kick the tires. Install it on your Kubernetes cluster by following [this guide](./example/README.md).
 
-## Table of contents
+To get a deeper understanding of how OSM does what it does - take a look at the detailed [software design](/DESIGN.md).
```
Comment (shashankram) [inline docs/development_guide.md:8]:
```suggestion
To get a deeper understanding of how OSM functions - take a look at the detailed [software design](/DESIGN.md).
```

Comment (draychev) [reply docs/development_guide.md:8]: Thanks!  Followed up here: https://github.com/openservicemesh/osm/pull/1460

**Pull request merged**